fvm/evm/precompiles/precompile.go

```diff
@@ -40,36 +40,93 @@ func MultiFunctionPrecompileContract(
 }
 
 type precompile struct {
-	address   types.Address
-	functions map[FunctionSelector]Function
+	address          types.Address
+	functions        map[FunctionSelector]Function
+	requiredGasCalls []types.RequiredGasCall
+	runCalls         []types.RunCall
 }
 
 func (p *precompile) Address() types.Address {
 	return p.address
 }
 
 // RequiredGas calculates the contract gas use
-func (p *precompile) RequiredGas(input []byte) uint64 {
+func (p *precompile) RequiredGas(input []byte) (output uint64) {
+	defer func() {
+		// lazy allocation
+		if p.requiredGasCalls == nil {
+			p.requiredGasCalls = make([]types.RequiredGasCall, 0)
+		}
+		p.requiredGasCalls = append(
+			p.requiredGasCalls,
+			types.RequiredGasCall{
+				Input:  input,
+				Output: output,
+			})
+	}()
 	if len(input) < FunctionSelectorLength {
-		return InvalidMethodCallGasUsage
+		output = InvalidMethodCallGasUsage
+		return
```

Comment:
I believe you don't have to do this, you can just use:
```
return InvalidMethodCallGasUsage
```
because of the named return

Reply:
[5d63313](https://github.com/onflow/flow-go/pull/6157/commits/5d6331332a79816d2787827de8e72a1e1646c2e7)